tests/components/mqtt/test_init.py

```diff
@@ -758,28 +758,54 @@ def mock_tls_set(certificate, certfile=None, keyfile=None, tls_version=None):
 )
 async def test_custom_birth_message(hass, mqtt_client_mock, mqtt_mock):
     """Test sending birth message."""
+    birth = asyncio.Event()
+
+    async def wait_birth(topic, payload, qos):
+        """Handle birth message."""
+        birth.set()
+
+    old_cooldown = mqtt.DISCOVERY_COOLDOWN
+    mqtt.DISCOVERY_COOLDOWN = 0.1
+    await mqtt.async_subscribe(hass, "birth", wait_birth)
     mqtt_mock._mqtt_on_connect(None, None, 0, 0)
     await hass.async_block_till_done()
+    await birth.wait()
     mqtt_client_mock.publish.assert_called_with("birth", "birth", 0, False)
+    mqtt.DISCOVERY_COOLDOWN = old_cooldown
 
 
 async def test_default_birth_message(hass, mqtt_client_mock, mqtt_mock):
     """Test sending birth message."""
+    birth = asyncio.Event()
+
+    async def wait_birth(topic, payload, qos):
+        """Handle birth message."""
+        birth.set()
+
+    old_cooldown = mqtt.DISCOVERY_COOLDOWN
+    mqtt.DISCOVERY_COOLDOWN = 0.1
```

Comment:
Use `patch.object('mqtt', 'DISCOVERY_COOLDOWN', 0.1)` so it is reverted after the test.